### PR TITLE
fix: geolocation api callbacks are not called

### LINKED ISSUE #11147: [Bug]-[2670]:geolocation callback is not called
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

https://docs.appsmith.com/framework-reference/appsmith#geolocation

Calling an API `onSuccess` of `getCurrentPosition` doesn't work.

```
{{
  appsmith.geolocation.getCurrentPosition(() => {
	  showAlert("ABC")
  })
}}
```

### Steps To Reproduce

1. On button click.
2. Call `appsmith.geolocation.getCurrentPosition`.
3. And call an API inside it's success callback.
4. Click the button.
5. Notice the API never gets called.

### workaround
Call the API after `getCurrentPosition` instead of the callback.
image

### Environment

Production

### Version

Cloud

### PR DESCRIPTION
**Pull request by ravikp7**
The success and error callbacks are not being called with `geolocation.getCurrentPosition` and errors were not handled properly with `geolocation.watchPosition`.
 
## Description
The success and error callbacks are now being called and the errors are handled correctly.

Fixes #11147 #9852

Media

https://user-images.githubusercontent.com/13567359/202650767-ea6f3dc0-8231-4abc-a586-c8ff0be54ef0.mp4

## Type of change

> Please delete options that are not relevant.

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?
- Manual
- Jest

### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [ ] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

File: app/client/src/sagas/ActionExecution/GetCurrentLocationSaga.ts [hunk 1]
```diff
@@ -118,8 +118,29 @@ export function* getCurrentLocationSaga(
     const currentLocation = extractGeoLocation(location);
 
     yield put(setUserCurrentGeoLocation(currentLocation));
+
+    if (actionPayload.onSuccess) {
+      yield call(executeAppAction, {
```
Comment (arunvjn) [inline app/client/src/sagas/ActionExecution/GetCurrentLocationSaga.ts:123]: Can you check what should happen when the success callback throws an error ?

Comment (ravikp7) [reply app/client/src/sagas/ActionExecution/GetCurrentLocationSaga.ts:123]: @arunvjn I just checked. In that case, the error callback is called as well as the error is shown in an alert.

Comment (arunvjn) [reply app/client/src/sagas/ActionExecution/GetCurrentLocationSaga.ts:123]: Could you please point me to the code that does this? AFAIK, `executeAppAction` doesn't throw Error.

**Review by arunvjn — APPROVED** [review]
Looks good!

**Comment by ravikp7** [comment]
@arunvjn I found an edge case issue. When the location access is turned off. The error callback is not being called, it's throwing an uncaught exception 1DOMException: Failed to execute 'postMessage' on 'Worker': GeolocationPositionError object could not be cloned.` 
I'm currently debugging this.

File: app/client/src/sagas/ActionExecution/GetCurrentLocationSaga.ts (hunk 1 above)
Comment (ravikp7) [reply app/client/src/sagas/ActionExecution/GetCurrentLocationSaga.ts:151]: There's `logActionExecutionError` statement for the alert.

**Comment by arunvjn** [comment]
@AnandiKulkarni Could please help verify this fix ?

**Review by arunvjn — DISMISSED** [review]

**Review by arunvjn — APPROVED** [review]

**Pull request merged**